.github/workflows/format-and-commit.yml

```diff
@@ -0,0 +1,37 @@
+name: Format And Commit
+
+concurrency:
+  group: ${{ github.ref }}
+
+on:
+    # Pending getting through individual PRs.
+    # push:
+    #   branches:
+    #     - master
+    workflow_dispatch: ~
+
+permissions:
+  contents: read
+
+jobs:
+  dprint-fmt:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
```

Comment:
We may want to clone this with the bot PR; pushing with the regular GHA token will not allow GHA to run on the commit, which may mask errors caused by formatting modifying `ExpectType` and `// @ts-expect-error`. I'm not sure if we have that secret available here, though; I don't have permission to access the settings for this repo. @andrewbranch or @sandersn may have perms to check?

Reply:
There’s `secrets.GH_DT_MERGEBOT_TOKEN`

Reply:
Yeah, if that exists, we should revert `contents: write` and clone with that token instead. The only way that can go wrong is if we push unformatted code and cause a loop (which is why you can't do this normally).

Reply:
```suggestion
    - uses: actions/checkout@v3
      with:
        token: ${{ secrets.GH_DT_MERGEBOT_TOKEN }}
```